File: shade-android-main/app/src/main/java/com/shade/app/data/remote/dto/GroupDto.kt
package com.shade.app.data.remote.dto

import com.google.gson.annotations.SerializedName

data class GroupMemberResponse(
    @SerializedName("user_id")  val userId: String,
    @SerializedName("shade_id") val shadeId: String,
    @SerializedName("role")     val role: String,
)

data class GroupResponse(
    @SerializedName("group_id")   val groupId: String,
    @SerializedName("name")       val name: String,
    @SerializedName("owner_id")   val ownerId: String,
    @SerializedName("avatar_url") val avatarUrl: String?,
    @SerializedName("members")    val members: List<GroupMemberResponse>,
    @SerializedName("created_at") val createdAt: String,
)

data class CreateGroupRequest(
    @SerializedName("name")       val name: String,
    @SerializedName("member_ids") val memberIds: List<String>,
)

data class AddMemberRequest(
    @SerializedName("user_id") val userId: String,
)

data class CreateInviteRequest(
    @SerializedName("group_id")  val groupId: String? = null,
    @SerializedName("max_uses")  val maxUses: Int = 1,
)

data class InviteResponse(
    @SerializedName("code")       val code: String,
    @SerializedName("group_id")   val groupId: String?,
    @SerializedName("max_uses")   val maxUses: Int,
    @SerializedName("use_count")  val useCount: Int,
    @SerializedName("expires_at") val expiresAt: String?,
)

data class RedeemInviteResponse(
    @SerializedName("type")    val type: String,          // "contact" | "group"
    @SerializedName("contact") val contact: LookupResponse?,
    @SerializedName("group")   val group: GroupResponse?,
)

File: shade-backend/internal/dto/group_dto.go
package dto

// ── Group requests ───────────────────────────────────────────────────────────

type CreateGroupRequest struct {
	Name      string   `json:"name"`       // required
	MemberIDs []string `json:"member_ids"` // list of user_id (UUID strings)
}

type AddMemberRequest struct {
	UserID string `json:"user_id"` // UUID string
}

// ── Group responses ──────────────────────────────────────────────────────────

type GroupMemberResponse struct {
	UserID  string `json:"user_id"`
	ShadeID string `json:"shade_id"`
	Role    string `json:"role"`
}

type GroupResponse struct {
	GroupID   string                `json:"group_id"`
	Name      string                `json:"name"`
	OwnerID   string                `json:"owner_id"`
	AvatarURL string                `json:"avatar_url,omitempty"`
	Members   []GroupMemberResponse `json:"members"`
	CreatedAt string                `json:"created_at"`
}

// ── Invite requests / responses ──────────────────────────────────────────────

type CreateInviteRequest struct {
	GroupID string `json:"group_id,omitempty"` // omit = personal contact invite
	MaxUses int    `json:"max_uses,omitempty"` // 0 → default 1
}

type InviteResponse struct {
	Code      string  `json:"code"`
	GroupID   *string `json:"group_id,omitempty"`
	MaxUses   int     `json:"max_uses"`
	UseCount  int     `json:"use_count"`
	ExpiresAt *string `json:"expires_at,omitempty"`
}

type RedeemInviteResponse struct {
	Type    string          `json:"type"`              // "contact" | "group"
	Contact *LookupResponse `json:"contact,omitempty"` // if type == "contact"
	Group   *GroupResponse  `json:"group,omitempty"`   // if type == "group"
}
